### DISCUSSION TITLE
I want to setViewPort for a selectedNode

### DISCUSSION THREAD

**Discussion by vinhphanybn1** [xyflow/xyflow#4853] [Q&A] [answered]
I want to setViewPort for a Node that I selected, but seems like it's wrong position.
I have tried to use the position of the Node, but it is still wrong now. 
This is my code: 

if (node?.position) {
    
      const realPosition = screenToFlowPosition(node.position)
      const nodeCenterX = realPosition.x + node.width / 2;
      const nodeCenterY = realPosition.y + node.height / 2;

      setViewport(
        {
          x: nodeCenterX,
          y: nodeCenterY,
          zoom: 1.3,
        },
        { duration: 800 }
      );
    }

**Comment by bcakmakoglu** [comment, marked as answer]
I think you're misunderstanding some things here

* `screenToFlowPosition` is meant to translate something like a mouse position on the screen to coordinates
	* translating a nodes position isn't going to work since it *already is a coordinate on the graph* and not a screen position
* `x` and `y` in `setViewport` will not align it *into the center*
	* it will align it so that x and y become the top-left coordinates of the viewport, so your node would never end up in the center using this

You can already achieve this way more easily by simply using `fitView` (as mentioned in your other comment in another discussion)

```ts
  const onNodeClick: NodeMouseHandler = (evt, node) => {
    fitView({
      nodes: [{ id: node.id }],
      duration: 800,
      minZoom: 1.3,
      maxZoom: 1.3,
    });
  };
``` 

[Here's an example.](https://codesandbox.io/p/sandbox/ecstatic-goodall-t98ngs?file=%2Fsrc%2FApp.tsx%3A29%2C1-36%2C5)

**Reply by vinhphanybn1** [reply]
Many thanks to you. It worked ^^
